woocommerce/woocommerce
# Check for empty value in get_order_id_by_order_key()

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:
Check for empty value in get_order_id_by_order_key()
This is especially necessary due to the function being able to call directly see https://github.com/woocommerce/woocommerce/issues/40055#issuecomment-1710260022

Reason is that with an empty variable the sql call gets prepared like this:
`SELECT post_id FROM wp_postmeta WHERE meta_key = '_order_key' AND meta_value = ''`

The issue author mentioned:
_I noticed a 3PD plugin wasn't checking for empty order key value before calling wc_get_order_id_by_order_key() which likely resulted in a slow query and made the site unresponsive. The site's postmeta table is considerably large._





Closes #40055 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note that this is performance change to prevent unnecessary query, so functionality isn't affected.

Codebase with PR

Start with posts authoritative:

Create a new order and note down it's order key, you can find it by clicking on the customer payment page in the order admin screen and then checking the URL -> &key=`wc_order_RHYQ7nzjy0Xd3`. It's also available on the URL bar in the thank you page after order checkout.
Then in shell -> 
1. call function **wc_get_order_id_by_order_key** with parameter $order_key (valid orderkey) -> should get the order ID
2. call function **wc_get_order_id_by_order_key** with _empty_ parameter $order_key (valid orderkey) -> should return 0 & query shouldn't run

Codebase with**out** PR
3. call function ** wc_get_order_id_by_order_key** with _empty_ parameter $order_key (valid orderkey) -> should return empty & query runs with empty orderkey





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Check if $order_key is empty in get_order_id_by_order_key() before running the sql query.

</details>
